cli/cmdlineparser.cpp

```diff
@@ -338,14 +338,16 @@ bool CmdLineParser::parseFromArgs(int argc, const char* const argv[])
             }
 
             // Exception handling inside cppcheck client
-            else if (std::strcmp(argv[i], "--exception-handling") == 0)
+            else if (std::strncmp(argv[i], "--exception-handling", 20) == 0) {
```

Comment:
this will now happily accept `--exception-handling-foo`

Reply:
Good catch! That also applies to `--clang` where I took the logic from. All other options are fine. Will fix for both.

Reply:
fixed.